packages/klighd-interactive/src/constraint-classes.ts

```diff
@@ -52,10 +52,11 @@ export class KNode extends RectangularNode implements KGraphElement {
     areChildAreaChildrenRendered = false
     areNonChildAreaChildrenRendered = false
     hasFeature(feature: symbol): boolean {
-        return feature === selectFeature || (feature === moveFeature && (this.parent as KNode).properties.interactiveLayout)
+        return feature === selectFeature || (feature === moveFeature && (this.parent as KNode).properties.get("org.eclipse.elk.interactiveLayout") as boolean)
     }
 
-    properties: NodeProperties
+    //properties: NodeProperties
+    properties: Map<string, unknown>
```

Comment:
You could throw a bit of Typescript's almighty type system on this map to make using it with known keys to not need the `as number` or similar casts every time a property is used outside of equality checks. See for example [this commit](https://github.com/kieler/vscode/commit/25ca28bf430ba31ec0bedb7c2daf49cde55ebd0f) on kieler/vscode.

Reply:
Having the keys of all properties as 'magic strings' may also not be ideal here and may cause bugs due to spelling errors. Having these as static variables to refer to somewhere or allow the type system to only allow correct keys as within the above commit could help here.